src/node_buffer.cc

```diff
@@ -743,14 +743,7 @@ void SlowByteLengthUtf8(const FunctionCallbackInfo<Value>& args) {
 
 uint32_t FastByteLengthUtf8(Local<Value> receiver,
                             const v8::FastOneByteString& source) {
-  uint32_t result = 0;
-  uint32_t length = source.length;
-  const uint8_t* data = reinterpret_cast<const uint8_t*>(source.data);
-  for (uint32_t i = 0; i < length; ++i) {
-    result += (data[i] >> 7);
-  }
-  result += length;
-  return result;
+  return simdutf::utf8_length_from_latin1(source.data, source.length);
```

Comment:
Can you also update line 741?

Reply:
Can you elaborate?

Line 741 calls into v8  (Utf8Length). That function flattens the content, and then branches out depending on whether the string is is Latin1 or UTF-8. But importantly, that's v8 code. It depends on several functions that are specific to v8 (and not used in the Node source code).

Do you want me to change code in v8? Or to bring into Node functions from v8 that are normally not used in Node?

This becomes a software engineering issue.

Reply:
I was thinking: We can call IsOneByte() and depending on the output call simdutf or V8 Utf8length.

Reply:
Something like this?

```C++
void SlowByteLengthUtf8(const FunctionCallbackInfo<Value>& args) {
  Environment* env = Environment::GetCurrent(args);
  CHECK(args[0]->IsString());
  auto v8_isolate = env->isolate();
  i::Handle<i::String> str = Utils::OpenHandle(args[0].As<String>());
  str = i::String::Flatten(reinterpret_cast<i::Isolate*>(v8_isolate), str);
  int length = str->length();
  if (length == 0) return 0;
  i::DisallowGarbageCollection no_gc;
  i::String::FlatContent flat = str->GetFlatContent(no_gc);
  DCHECK(flat.IsFlat());
  int utf8_length = 0;
  if (flat.IsOneByte()) {
    auto source = flat.ToOneByteVector();
    return simdutf::utf8_length_from_latin1(source.data, source.length);
  } else {
    auto source = flat.ToUC16Vector();
    return simdutf::utf8_length_from_utf16(source.data, source.length);
  }
}
```

Reply:
Yes! @joyeecheung wdyt?

Reply:
We cannot (or are not supposed to) access anything in `v8::internal`, so the pieces needing access to V8 internals needs to land in the upstream and be exposed as public API.